### PR TITLE
Port DataflowJavaSDK-351 to Beam

### PR DESCRIPTION
**Pull request by peihe**
Port PR "Check for existence of job before polling." to Beam
https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/351

### CODE REVIEW

**Comment by peihe** [comment]
R: @lukecwik

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java [hunk 1]
```diff
@@ -283,6 +283,51 @@ public JobStatistics dryRunQuery(String projectId, String query)
           Sleeper.DEFAULT,
           backoff).getStatistics();
     }
+    /**
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: Missing new line above start of comment

**Comment by lukecwik** [comment]
LGTM

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java (hunk 1 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: done

**Comment by peihe** [comment]
fixed, PTAL

Thanks Luke!

**Pull request closed without merging**